Storage/ThirdParty/SwiftData.swift

```diff
@@ -113,16 +119,18 @@ public class SwiftData {
             conn = ConcreteSQLiteDBConnection(filename: filename, flags: flags.toSQL(), key: self.key, prevKey: self.prevKey)
         }
 
-        // Initialize in case we encounter a nil connection when we go to run the dispatch blocks
-        let failed = FailedSQLiteDBConnection()
         let queue = self.sharedConnectionQueue
-        let noConnection = NSError(domain: "mozilla", code: 0, userInfo: [NSLocalizedDescriptionKey: "Could not create a connection"])
-
         if synchronous {
             var error: NSError? = nil
             dispatch_sync(queue) {
+                /**
+                 * By the time this dispatch block runs, it is possible the user has backgrounded the app
+                 * and the connection has been dealloc'ed since we last grabbed the reference
```

Comment:
Please do the small amount of mental legwork to be assured that if I do this:

```
   long write X
   write A
   write B
   force close
   write C
   reopen
```

some consistent collection of writes is performed — e.g., none of A, B, C, or only C, or all of A, B, C.